lib/rules.js

```diff
@@ -184,6 +227,185 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
+
+// e.g. nthRoot(2 * x, 2) -> nthRoot(2, 2) * nthRoot(x, 2)
+export const DISTRIBUTE_NTH_ROOT = defineRuleString('nthRoot(#a_0 * ..., #b)', 'nthRoot(#a_0, #b) * ...')
+
+
+// TODO: #10 and #23 (`or` rules and applying multiple rules if
+// multiple occurrences)
+// e.g. nthRoot(4, 2) * nthRoot(x^2, 2) -> 2 * x
+export const EVALUATE_DISTRIBUTED_NTH_ROOT = defineRule(
+    (node) => {
+        let isDistributed = query.isMul(node)
+        let canEvaluate = false
+        if (isDistributed) {
+            canEvaluate = node.args.every(arg => canApplyRule(CANCEL_EXPONENT, arg) || canApplyRule(NTH_ROOT_VALUE, arg))
+        }
+        return (isDistributed && canEvaluate) ? {node} : null
+    },
+
+    (node) => {
+        const result = build.applyNode(
+            'mul',
+            node.args.map(nthRoot => {
+                let radicand = nthRoot.args[0]
+                let index = nthRoot.args[1]
+                console.log(index)
+                if (query.isNumber(radicand)) {
+                    return applyRule(NTH_ROOT_VALUE,
+                                     build.applyNode(
+                                         'nthRoot',
+                                         [radicand, index]
+                                     ))
+                } else {
+                    return applyRule(CANCEL_EXPONENT,
+                                     build.applyNode(
+                                         'nthRoot',
+                                         [radicand, index]
+                                     ))
+                }
+            })
+        )
+        return result
+    }
+)
+
+// e.g. 12 -> 2 * 2 * 3
+export const FACTOR_INTO_PRIME = defineRule(
+    (node) => {
+        return query.isNumber(node) ? {node} : null
+    },
+
+    (node) => {
+        const factors = primeFactorization(query.getValue(node))
+        const result = build.applyNode(
+            'mul',
+            factors.map(build.numberNode)
+        )
+        return result
+    }
+)
+
+// e.g. nthRoot(2 * 2 * 2, 2) -> nthRoot((2 * 2) * 2)
+export const GROUP_TERMS_BY_ROOT = defineRule(
+    (node) => {
+        let canGroupTerms = false
+        canGroupTerms = canApplyRule(DISTRIBUTE_NTH_ROOT, node)
+        return canGroupTerms ? {node} : null
+    },
+
+    (node) => {
+        const radicand = node.args[0]
+        const index = node.args[1]
+
+        // dictionary storing the number of times a constant appears
+        // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
+        const count = {}
+        radicand.args.forEach(arg => {
+            let key = JSON.stringify(arg)
+            count[key] ? count[key]++ : count[key] = 1
+        })
+
+        const flatten = arr => arr.reduce(
+            (acc, val) => acc.concat(
+                Array.isArray(val) ? flatten(val) : val
+            ),
+            []
+        )
```

Comment:
Interesting... `Object.keys(count).map(key => { ... }` should be returning an array of arrays since each iteration returns `args` which is an array.  None of the items in `args` should be arrays themselves so flattening shouldn't be necessary.